### ISSUE TITLE
Missing ARIA on nimble-table

### ISSUE THREAD

**Issue by fredvisser** [ni/nimble#2285]
# 🐛 Bug Report

Per the [ARIA Treegrid example](https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/examples/treegrid-1/) and the [ARIA APG Grid and Table Properties](https://www.w3.org/WAI/ARIA/apg/practices/grid-and-table-properties/) documents, we are missing a few ARIA properties required to fully describe the nimble-table for screenreaders.

Missing label | Role | Description 
-- | -- | --
aria-label="Inbox" | treegrid | Provides an accessible name for the treegrid.
aria-level="number" | row | Defines the level of the row in the hierarchical treegrid structure. Counting is one-based. Root rows have aria-level=“1”.
aria-setsize="number"	| row |	Defines the number of rows in the set of rows that are in the same branch and at the same level within the hierarchy.
aria-posinset="number" | row | Defines the position of the row within the set of other rows that are in the same branch and at the same level within the hierarchy. Counting is one-based, not zero-based.
Change role="cell" to role="gridcell" | cell | Identifies the element as a gridcell. The gridcell role is not an implicit semantic for the tdelement when in a treegrid.

Property | Definition
-- | --
aria-colcount |	Defines the total number of columns in a table, grid, or treegrid.
aria-rowcount |	Defines the total number of rows in a table, grid, or treegrid.
aria-colindex |	Defines a cell's position with respect to the total number of columns within a table, grid, or treegrid. Note: Numbering starts with 1, not 0.
aria-rowindex |	Defines a cell's position with respect to the total number of rows within a table, grid, or treegrid. Note: Numbering starts with 1, not 0.
